contour/src/main/kotlin/com/squareup/contour/solvers/SimpleAxisSolver.kt

```diff
@@ -116,30 +133,42 @@ internal class SimpleAxisSolver(
 
   override fun range(): Int {
     if (range == Int.MIN_VALUE) {
-      parent.measureSelf()
+      resolveRange()
     }
     return range
   }
 
+  private fun resolveRange() {
+    if (parent.view.visibility == View.GONE) {
+      onRangeResolved(0, 0)
+    } else {
+      if (p1.isSet && p1.mode == SizeMode.Exact) {
```

Comment:
Could we remove all the instances of:
```
if (p0.point == Point.Baseline && baselineRange == Int.MIN_VALUE) {
  parent.measureSelf()
}
```
from above, and move that block into the first condition of `else-if` chain?

Reply:
No. It's the same as removing the code completely. Because `p0.point == Point.Baseline` and `p1.isSet` are mutually exclusive. `baselineTo` returns `HeightOfOnlyContext` which does not allow setting `p1`, only `size` can be set.

I was expecting that to break the tests but it didn't. I've written a failing test case but I'm currently blocked by #118.

Reply:
Reading my comment, I am realizing i was not very clear 😑. 

I think we could clean up the by moving the block:
```
if (p0.point == Point.Baseline && baselineRange == Int.MIN_VALUE) {
  parent.measureSelf()
}
```
to here:
```
private fun resolveRange() {
    if (parent.view.visibility == View.GONE) {
      onRangeResolved(0, 0)
    } else {
      if (p0.point == Point.Baseline && baselineRange == Int.MIN_VALUE) {
        parent.measureSelf()
      } else if (p1.isSet && p1.mode == SizeMode.Exact) {
        range = abs(p0.resolve() - p1.resolve())
      } else if (size.isSet && size.mode == SizeMode.Exact) {
        range = size.resolve()
      } else {
        parent.measureSelf()
      }
    }
  }
```